Tests/Fixtures/Generated/Strings/flat-swift4/plurals-advanced.swift

```diff
@@ -0,0 +1,45 @@
+// swiftlint:disable all
+// Generated using SwiftGen — https://github.com/SwiftGen/SwiftGen
+
+import Foundation
+
+// swiftlint:disable superfluous_disable_command file_length implicit_return
+
+// MARK: - Strings
+
+// swiftlint:disable function_parameter_count identifier_name line_length type_body_length
+internal enum L10n {
+  /// Plural case 'other': %@ has %d ratings
+  internal static func multiplePlaceholdersStringInt(_ p1: Int) -> String {
```

Comment:
As I explained [above](https://github.com/SwiftGen/SwiftGen/pull/634#issuecomment-633491998), I don't think this is a bug. IMO this is the intended behavior of the `.stringsdict` file format. Due to NSLocalizedString probably just replacing all `CVarArgs` being thrown into the variadic parameter, it could be seen as a limitation that SwiftGen does not parse this `%@` placeholder _that is not an explicit `variable`_ of the format key of the plural. 

I'd suggest to also document this behavior instead of trying to fix it, as this would require parsing all variable rules (zero, one, many, other, ...) and could result in different parameters for each of them. If this kind of String is desired, it needs to declare all placeholders in the format key, or it should be used as a combination of `%@ has` in `Localizable.strings` and `%d ratings` in `Localizable.stringsdict`.

Reply:
Ok I re-read your initial comment and I think I start to understand better now.

But I'm also a bit doubtful of the documentation because I see some inconsistencies inside the doc and between the doc and reality. So I think it deserves some more practical tests to validate all that instead of only relying on the doc.

For example, the doc says:

> `NSStringFormatValueTypeKey`
>   A string format specifier for a number, as in %d for an integer. 

But `NSStringFormatValueTypeKey` can in practice take a format specifying for other than a number, at least according to the examples you use in that PR (like [here](https://github.com/SwiftGen/SwiftGen/pull/634/files#diff-b878e878c299da6865383bc7df21c027R149-R150)). So which is it, is the documentation right and we should only always use placeholders for numbers only, or are we also allowed to use placeholders like `@`? That's one inconsistency between doc and reality


Also, I think I also saw cases IRL where the `NSStringLocalizedFormatKey` had a mixed of placeholders-with-variable-name and regular-placeholders, only using the ones with variable for the placeholders that needs to apply the plural rules, and regular ones for values invariants to plural rules.

For example `"%@ %#@has_rating@"` could be a valid `NSStringLocalizedFormatKey` in practice (even if the documentation doesn't mention that explicitly in theory), where the rest of the dict would only contain an entry for the `has_ratings` variable, and the behaviour at runtime would be to expect a `String` for the first `%@` placeholder and an `Int` for the second, variable-based placeholder.

I think it's a perfect occasion to do some TDD there, write some test cases for edge cases like that we could think about in our twisted minds (😈), then see what's generated by SwiftGen for those… but also while we're working on this PR, test that generated code in a playground or dummy project on the side to check the behaviour of those edge cases at runtime (instead of entirely relying on the documentation which seems not completely exhaustive to me) to validate our theory

It would also be interesting to use one of the common tools like Lokalise or similar to see what's exported by those tools when you use a plural with multiple placeholders for a given entry.